overlord/state/taskrunner.go

```diff
@@ -267,6 +267,8 @@ func (r *TaskRunner) run(t *Task) {
 			r.abortLanes(t.Change(), t.Lanes())
 			t.SetStatus(ErrorStatus)
 			t.Errorf("%s", err)
+			// ensure the error is available in the global log too
+			logger.Noticef("task %q failed with: %v", t.Summary(), err)
```

Comment:
so my plan from 2016 was that all the task logging would go also to the journal with some prefixing, I think it's still a good idea, but the plan also involved introducing logger.Errorf which is a fair bit of work.  Anyway with that plan there would be nothing almost to do here, but as a temporary stop gap we should try to log something like what the plan would likely log:

```
[change %s %q task] failed: %v
```

using the change ID of the task, the task Summary and the err

Reply:
Perhaps we could use rich jorunal logging APIs that allow us to essentially publish a json dict along with each message. http://0pointer.de/blog/projects/journal-submit.html contains the details and the bigger ideas.

This would allow administrators (and us) to ask for precise messages to look for particular behavior.

Reply:
that is its own mini project (also we would need to think through its relation to a potential event stream API for snapd which people have also requested)

Reply:
With this change the journal output for a failed install is:
```
taskrunner.go:217: [change 2 "Setup system for run mode" task] failed: cannot create parition: gadget ... 
```
which looks pretty nice IMO

Reply:
Updated via force-push (for easier cherry pick) and added a small test